### PR TITLE
Add xkit post

### PR DESCRIPTION
**Pull request by mooreds**

### CODE REVIEW

File: site/_posts/2020-12-08-xkit-and-fusionauth.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+layout: blog-post
+title: Adding native integrations to your app with FusionAuth and Xkit
+description: How to use FusionAuth and Xkit two services together to simplify auth and integration development.
```
Comment (voidmain) [inline site/_posts/2020-12-08-xkit-and-fusionauth.md:4]: "two services" sounds odd. Should that just "services"?

File: site/_posts/2020-12-08-xkit-and-fusionauth.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+
+## What is Xkit
+
+Xkit is a SaaS platform which makes integrating third party systems a snap. Suppose you are writing a recipe management application and are going to sell it for big money to all the cooks of the world. After some market research, you realize that you want to integrate with other services. Your users are clamoring for the ability to export the steps of a recipe to a Trello board for sharing and Dropbox for backups. These are all services with APIs. 
+
+End users can give your application access to their accounts with these services, but that takes some coding. There's also a fair bit of hoop jumping: setting up API keys and OAuth consent screens, among other things.
+
+This is the problem which Xkit solves. If Xkit has a connection ([here's a list](https://docs.xkit.co/docs/connecting-with-apps-overview) of supported services), your application can, after Xkit is installed configured, allow a user to connect to a service. I (Dan) was able to connect to Trello to my app in about an hour. The user experience of connecting the external application is smooth and far better than something I could whip up in a day, let alone an hour.
```
Comment (voidmain) [inline site/_posts/2020-12-08-xkit-and-fusionauth.md:25]: "to Trello to my" should probably be "Trello to my"

File: site/_posts/2020-12-08-xkit-and-fusionauth.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+## Xkit and FusionAuth integration
+
+### Install FusionAuth
+
+FusionAuth offers [a number of different methods you can use to install the service](https://fusionauth.io/docs/v1/tech/installation-guide/). Once you've installed, there's a Setup Wizard to walk you through the next steps. You'll need to create your application in the FusionAuth interface and then add a few elements to your application code base to fully implement the FusionAuth login flow. A [full, detailed setup guide](https://fusionauth.io/docs/v1/tech/5-minute-setup-guide/) is also available. Feel free to create additional [users via the FusionAuth administrative user interface](https://fusionauth.io/docs/v1/tech/core-concepts/users/).
```
Comment (voidmain) [inline site/_posts/2020-12-08-xkit-and-fusionauth.md:31]: Should "Once you've installed" be "Once you've installed it"?

File: site/_posts/2020-12-08-xkit-and-fusionauth.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+Once this is installed you have a full featured user management system, ready to go. APIs to control everything, multi factor authentication, consent management, SAML, OIDC, and more, hosted wherever you want. There's also FusionAuth cloud, a managed services offering, if you don't want to host FusionAuth yourself.
```
Comment (voidmain) [inline site/_posts/2020-12-08-xkit-and-fusionauth.md:33]: I'd reword this to "Once FusionAuth is installed and configured, you have a full featured user management system, ready to go."

Comment (voidmain) [inline site/_posts/2020-12-08-xkit-and-fusionauth.md:33]: We also usually capitalize "FusionAuth Cloud" since it is a service name.

File: site/_posts/2020-12-08-xkit-and-fusionauth.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+### Connect FusionAuth with Xkit
+
+Now that you have both FusionAuth and Xkit set up, you'll need to connect the two. We do this by collecting some information from your FusionAuth dashboard and inputting it into Xkit. 
+
+Specifically, you'll need to generate an RSA key and then add your "iss" claim, "aud" claim and JWKS URL into your Xkit account. This setup is [fully documented](https://docs.xkit.co/docs/fusionauth).
+
+{% include _image.liquid src="/assets/img/blogs/xkit-fusionauth-integration/rsa-keypair.png" alt="The generated RSA key for use with Xkit." class="img-fluid" figure=false %}
+
+Once you've done that and clicked save, FusionAuth and Xk9it will be connected. Huzzah!
```
Comment (voidmain) [inline site/_posts/2020-12-08-xkit-and-fusionauth.md:60]: Looks like a typo in "Xk9it".

File: site/_posts/2020-12-08-xkit-and-fusionauth.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+You can now have your user log in using FusionAuth's Login API, federated with Active Directory, Google or Facebook, or an OAuth authorization code flow. You'll get a JWT from FusionAuth on successful user authentication. This JWT can then be sent to Xkit to authenticate the user in Xkit, and therefore grant them access to integrations you've configured.
```
Comment (voidmain) [inline site/_posts/2020-12-08-xkit-and-fusionauth.md:62]: Should "You can now have your user log in using FusionAuth's Login API" be "You can now have your users log in using FusionAuth's Login API" (with the s on user)?

**Review by voidmain — COMMENTED** [review]
Left a few comments. I didn't run this in the browser though, so you'll want to verify all the links and images are correct.

File: site/_posts/2020-12-08-xkit-and-fusionauth.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+title: Adding native integrations to your app with FusionAuth and Xkit
+description: How to use FusionAuth and Xkit two services together to simplify auth and integration development.
+image: blogs/xkit-fusionauth-integration/adding-native-integrations-to-your-app-with-fusionauth-and-xkit-header-image.png
+author: Trey Griffith and Dan Moore
+category: blog
+tags: client-javascript 
+excerpt_separator: "<!--more-->"
+---
+
+FusionAuth and Xkit came together for this blog post to share how you can seamlessly use our two services together to boost your engineering team's productivity. If you're working on growing your SaaS business, you know just how much your engineers have on their plates. At both FusionAuth and Xkit, we believe that outsourcing what you can -- like authentication and integration infrastructure -- lets your team focus on the core parts of your business that will really make it a success.
```
Comment (dpolstra) [inline site/_posts/2020-12-08-xkit-and-fusionauth.md:12]: "how you can use our services"

Comment (dpolstra) [inline site/_posts/2020-12-08-xkit-and-fusionauth.md:12]: lets your team focus on the products and services that drive your business.

File: site/_posts/2020-12-08-xkit-and-fusionauth.md (hunk 2 above)
Comment (dpolstra) [inline site/_posts/2020-12-08-xkit-and-fusionauth.md:25]: ~after Xkit is installed and configured~

File: site/_posts/2020-12-08-xkit-and-fusionauth.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+//...
+xkit.ready(() => {
+  xkit.login('eyJhbGciOi...')
+})
+//...
+```
+						
+This easy JWT-based connection saves you the trouble of dealing with API keys and provisioning users for Xkit; you can instead maintain them in FusionAuth. You can also use FusionAuth for all your other applications, providing one view of all your users.
+						
+The security minded among you will notices that this JWT is available in the DOM, and therefore exposed to cross site scripting attacks, should any malicious JavaScript be executed on the same page. To minimize the risks, lock the permissions associated with this JWT down and don't allow its use as a bearer token for any other more sensitive APIs or services.
```
Comment (dpolstra) [inline site/_posts/2020-12-08-xkit-and-fusionauth.md:76]: "minded among you will notice ~s~"
